src/main/java/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -78,10 +81,24 @@ public void start() {
     super.start();
     // we may start multiple times due to loading after saving
     // only initialize once
+    final PlayerID player = m_bridge.getPlayerID();
     if (m_needToInitialize) {
       doInitialize(m_battleTracker, m_bridge);
+      needToFindRocketTargets = true;
+      needToFireRockets = true;
       m_needToInitialize = false;
     }
+    // WW2V2/WW2V3, fires at end of combat move - now moved to the start of the BattleDelegate
+    // WW2V1, fires at end of non combat move
+    if (GameStepPropertiesHelper.isFireRockets(getData()) && TechTracker.hasRocket(player)) {
+      if (needToFindRocketTargets && !games.strategy.triplea.Properties.getStrictRockets(getData())) {
+        rocketHelper.findRocketTargets(m_bridge, player);
```

Comment:
For example, with the state in rocketHelper, by also passing the game data, all of this new logic here seemingly can be replaced with: ` rocketHelper.findRocketTargets(m_bridge, player, getData());`